lib/Target/Cpp/TranslateToCpp.cpp

```diff
@@ -49,6 +49,32 @@ static LogicalResult printConstantOp(CppEmitter &emitter,
   return success();
 }
 
+static LogicalResult printConstOp(CppEmitter &emitter, ConstOp constOp) {
+  auto &os = emitter.ostream();
+  if (failed(emitter.emitType(constOp.getType())))
+    return failure();
+  os << " " << emitter.getOrCreateName(constOp.getResult());
+
+  // Add braces for number literals only to avoid double brace intialization for
+  // tensors.
+  auto value = constOp.value();
+  bool emitBraces = value.isa<FloatAttr>() || value.isa<IntegerAttr>();
+
+  if (emitBraces)
+    os << "{";
+
+  if (constOp.getType().dyn_cast<emitc::OpaqueType>())
+    os << " = ";
```

Comment:
This goes wrong and prints something like `v1 { = ...}` when emitBraces is also true. We could emit an error in this case.

Reply:
I have addressed this with an additional commit. We should refactor a little larger in future, but I think that's fine for now.